Indicators/TimeSeriesForecast.cs

```diff
@@ -0,0 +1,89 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+namespace QuantConnect.Indicators
+{
+/// <summary>
+/// Represents an indicator capable of predicting new values given previous data from a window.
+/// Source: https://tulipindicators.org/tsf
+/// </summary>
+public class TimeSeriesForecast : WindowIndicator<IndicatorDataPoint>, IIndicatorWarmUpPeriodProvider
+{
+    /// <summary>
+    /// Gets a flag indicating when this indicator is ready and fully initialized
+    /// </summary>
+    public override bool IsReady => Samples >= Period;
+
+    /// <summary>
+    /// Required period, in data points, for the indicator to be ready and fully initialized
+    /// </summary>
+    public int WarmUpPeriod => Period;
+    
+    /// <summary>
+    /// Creates a new TimeSeriesForecast indicator with the specified period
+    /// </summary>
+    /// <param name="name">The name of this indicator</param>
+    /// <param name="period">The period over which to look back</param>
+    public TimeSeriesForecast(string name, int period)
+        : base(name, period)
+    {
+    }
+
+    /// <summary>
+    /// Creates a new TimeSeriesForecast indicator with the specified period
+    /// </summary>
+    /// <param name="period">The period over which to look back</param>
+    public TimeSeriesForecast(int period)
+        : base($"TSF{period})", period)
+    {
+    }
+
+    /// <inheritdoc />
```

Comment:
Sorry our parsing documentation tech doesn't support `inheritdoc`, please use the complete expanded version